src/useSubtitles.js

```diff
@@ -0,0 +1,114 @@
+import { getVoiceFromCue, isJSON, toIterable } from './machinery/utilities';
+import { initial, initialCurrentSubtitle } from './defaults';
+import { useCallbackRef } from "./machinery/useCallbackRef";
+import { useEvent } from "./machinery/useEvent";
+
+export function useSubtitles({ language = "nl" }) {
+  const [subtitles, setSubtitles] = React.useState({ [language]: [] });
+  const [metadata, setMetadata] = React.useState({ [language]: [] });
+  const [currentSubtitle, setCurrentSubtitle] = React.useState({ [language]: initialCurrentSubtitle });
+  const [currentMetadata, setcurrentMetadata] = React.useState({ [language]: initial });
+
+  const onMetadataChangeEvent = useEvent(handleMetadataChange);
+  const onCueChangeEvent = useEvent(handleCueChange);
+  const onMountEvent = useEvent(handleInitialLoad);
+  const onUnmountEvent = useEvent(handleUnmount);
+
+  const memoizedMetadata = React.useMemo(() => metadata, [metadata]);
+  const memoizedSubtitles = React.useMemo(() => subtitles[language], [subtitles, language]);
+  const memoizedCurrentMetadata = React.useMemo(() => currentMetadata[language], [currentMetadata, language]);
+  const memoizedCurrentSubtitle = React.useMemo(() => currentSubtitle[language], [currentSubtitle, language])
+
+  const ref = useCallbackRef({
+    onMount: onMountEvent,
+    onUnmount: onUnmountEvent
+  })
+
+  /** @param {HTMLMediaElement} node */
+  function handleInitialLoad(node) {
+    const tracks = toIterable(node?.textTracks)
+
+    // Set initial subtitles
+    node?.addEventListener('loadedmetadata', () => {
+      tracks.forEach((x) => {
+        const hasSubtitles = subtitles[language].length
+        const hasMetadata = metadata[language].length
+
+        if (!hasSubtitles && x.kind === 'subtitles') {
+          setInitialSubtitles(x)
+        }
+        
+        if (!hasMetadata && x.kind === 'metadata') {
+          setInitialMetadata(x)
+        }
+      })
+    })
+
+    // Handle cue changes
+    tracks.forEach((x) => {
+      if (x.kind === 'subtitles') {
+        x.addEventListener("cuechange", onCueChangeEvent);
+      }
+      
+      if (x.kind === 'metadata') {
+        x.addEventListener("cuechange", onMetadataChangeEvent);
+      }
+
+      x.mode = "hidden";
```

Comment:
Wat doet dit precies?

Reply:
Ik zou dit misschien aan de gebruiker overlaten: als die een eigen implementatie wil maken, kan het `default` attribuut weglaten (of krijg je dan geen cuechange events meer?)

Reply:
Bij nader inzien had ik dit toch nodig; als je geen `mode = hidden` meegeeft, word de track niet blootgesteld aan javascript, en krijg je geen dus data.